rest/rest-api/src/main/java/org/ow2/proactive_grid_cloud_portal/dataspace/dto/ListFile.java

```diff
@@ -38,35 +38,34 @@
 
 import org.codehaus.jackson.map.annotate.JsonSerialize;
 
+import java.util.ArrayList;
+import java.util.Collections;
 import java.util.List;
 
 
 @JsonSerialize(include = JsonSerialize.Inclusion.NON_EMPTY)
 public class ListFile {
-    private List<String> directoryListing;
-    private List<String> fileListing;
-    private List<String> fullListing;
+    private final List<String> directoryListing;
+    private final List<String> fileListing;
+    private final List<String> fullListing;
 
-    public List<String> getDirectoryListing() {
-        return directoryListing;
+    public ListFile(List<String> directoryListing, List<String> fileListing, List<String> fullListing) {
+        this.directoryListing = new ArrayList<>(directoryListing);
+        this.fileListing = new ArrayList<>(fileListing);
+        this.fullListing = new ArrayList<>(fullListing);
+        Collections.sort(this.directoryListing);
+        Collections.sort(this.fileListing);
+        Collections.sort(this.fullListing);
     }
 
-    public void setDirectoryListing(List<String> directoryListing) {
-        this.directoryListing = directoryListing;
+    public List<String> getDirectoryListing() {
+        return directoryListing;
```

Comment:
Can you return a copy of the directoryListing to prevent the caller from getDirectoryListing to access (and possible change) the sorted List.